google-cloud-dialogflow/MIGRATING.md

```diff
@@ -0,0 +1,274 @@
+choose to install only one of the versioned gems. For example, if you know you
+will only use `V2` of the service, you can install `google-cloud-dialogflow-v2`
+by itself, and construct instances of the
+`Google::Cloud::Dialogflow::V2::Agents::Client` client class directly.
+
+### Client Configuration
+
+In older releases, if you wanted to customize performance parameters or
+low-level behavior of the client (such as credentials, timeouts, or
+instrumentation), you would pass a variety of keyword arguments to the client
+constructor. It was also extremely difficult to customize the default settings.
+
+With the 1.0 release, a configuration interface provides access to these
+parameters, both global defaults and per-client settings. For example, to set
+global credentials and default timeout for all Dialogflow V2 sessions clients:
+
+```
+Google::Cloud::Dialogflow::V2::Sessions::Client.configure do |config|
+  config.credentials = "/path/to/credentials.json"
+  config.timeout = 10_000
+end
+```
+
+Individual RPCs can also be configured independently. For example, to set the
+timeout for the `detect_intent` call:
+
+```
+Google::Cloud::Dialogflow::V2::Sessions::Client.configure do |config|
+  config.rpcs.detect_intent.timeout = 20_000
+end
+```
+
+You can also set certain configuration defaults for all Dialogflow versions and
+services globally:
+
+```
+Google::Cloud::Dialogflow.configure do |config|
+  config.credentials = "/path/to/credentials.json"
+  config.timeout = 10_000
+end
+```
+
+### Creating Clients
+
+In older releases, to create a client object, you would use the `new` method
+of modules under `Google::Cloud::Dialogflow`. For example, you might call
+`Google::Cloud::Dialogflow::Agents.new`. Keyword arguments were available to
+select a service version and to configure parameters such as credentials and
+timeouts.
+
+Wiht the 1.0 release, use named class methods of `Google::Cloud::Dialogflow` to
+create a client object. For example, `Google::Cloud::Dialogflow.sessions()`.
+You may select a service version using the `:version` keyword argument.
+However, other configuration parameters should be set in a configuration block
+when you create the client.
+
+Old:
+```
+client = Google::Cloud::Dialogflow::Agents.new credentials: "/path/to/credentials.json"
+```
+
+New:
+```
+client = Google::Cloud::Dialogflow.agents do |config|
+  config.credentials = "/path/to/credentials.json"
+end
+```
+
+The configuration block is optional. If you do not provide it, or you do not
+set some configuration parameters, then the default configuration is used. See
+[Client Configuration](#client-configuration).
+
+### Passing Arguments
+
+In older releases, certain required arguments would be passed as positional
+arguments, while most optional arguments would be passed as keyword arguments.
+
+With the 1.0 release, all RPC arguments are passed as keyword arguments,
+regardless of whether they are required or optional. For example:
```

Comment:
If true, add: "with documentation that specifies whether they are required or optional."

Reply:
Not necessarily true.